plasmapy/particles/particle_collections.py

```diff
@@ -0,0 +1,207 @@
+"""Collections of `~plasmapy.particles.particle_class.Particle` objects."""
+
+__all__ = ["ParticleList"]
+
+import astropy.units as u
+import collections
+import numpy as np
+
+from typing import *
+
+from plasmapy.particles.decorators import particle_input
+from plasmapy.particles.exceptions import *
+from plasmapy.particles.particle_class import (
+    CustomParticle,
+    DimensionlessParticle,
+    Particle,
+    ParticleLike,
+)
+
+
+class ParticleList(collections.UserList):
+    """
+    A list-like collection of `Particle` and/or `CustomParticle` objects.
+
+    Parameters
+    ----------
+    particles : iterable
+        An iterable that provides a sequence of `ParticleLike` objects.
+
+    Examples
+    --------
+    >>> from plasmapy.particles import ParticleList
+    >>> particle_list = ParticleList(["e-", "e+"])
+    >>> particle_list[0]
+    Particle("e-")
+    >>> particle_list.mass
+    <Quantity [9.1093...e-31, 9.1093...e-31] kg>
+    >>> particle_list.charge
+    <Quantity [-1.60217663e-19,  1.60217663e-19] C>
+    >>> particle_list.symbols
+    ['e-', 'e+']
+    >>> particle_list + [Particle("p+")]
+    ParticleList(['e-', 'e+', 'p+'])
+    """
+
+    @staticmethod
+    def _list_of_particles_and_custom_particles(
+        particles: Iterable[ParticleLike],
+    ) -> List[Union[Particle, CustomParticle]]:
```

Comment:
Wow, now I wish we had https://github.com/PlasmaPy/PlasmaPy/issues/687 for this particular annotation. :sweat_smile: 

In fact, what do you think about tagging this...

```suggestion
    ) -> List[Union[Particle, CustomParticle]]:  # TODO #687
```

I can probably set up a grep-by-issue-numbers script to list such notes.